src/net_processing.cpp

```diff
@@ -784,6 +784,77 @@ void Misbehaving(NodeId pnode, int howmuch, const std::string& message)
         LogPrint(BCLog::NET, "%s: %s peer=%d (%d -> %d)%s\n", __func__, state->name, pnode, state->nMisbehavior-howmuch, state->nMisbehavior, message_prefixed);
 }
 
+/**
+ * Returns true if the given validation state result may result in us banning/disconnecting a peer
+ * which provided such an object. This is used to determine whether to relay transactions to
+ * whitelisted peers, preventing us from relaying things which would result in them disconnecting
+ * us.
+ */
+static bool MayResultInDisconnect(const CValidationState& state, bool via_compact_block) {
+    switch (state.GetReason()) {
+    case ValidationInvalidReason::NONE:
+        return false;
+    // The node is is providing invalid data:
+    case ValidationInvalidReason::CONSENSUS:
+    case ValidationInvalidReason::MUTATED:
+    case ValidationInvalidReason::UNKNOWN_INVALID: // Really "duplicate of invalid "
```

Comment:
"where the enum is defined" - I'm confused, at the enum definition it says "this object was cached as being invalid, but we don't know why", is that insufficient?

Reply:
Apparently it is insufficient, since you've felt the need to leave a comment here explaining what you really mean (and repeated it in the function below) :slightly_smiling_face:

Reply:
Agreed the comment should be carried over, or the value renamed.

Reply:
Cleaned up.